src/main/java/core/basesyntax/Rectangle.java

```diff
@@ -16,6 +16,27 @@ public Rectangle(Integer width, Integer length, String color) {
         this.color = color;
     }
 
+    @Override
+    public int hashCode() {
+        return getByteSum(this.color) * this.getWidth() * this.getLength();
+    }
+
+    @Override
+    public boolean equals(Object obj) {
+        if (this == obj) {
+            return true;
+        }
+        if (obj != null && this.getClass() == obj.getClass()) {
+            Rectangle rectObj = (Rectangle)obj;
+            if (this.width - rectObj.getWidth() == 0) {
```

Comment:
What happen in we change `Integer width` to `String width`?
Is it correct to check difference of the values?